syntax = "proto2";

package e2e_noa;

enum TurnSignal {
  TURN_SIGNAL_NONE = 0;
  TURN_SIGNAL_LEFT = 1;
  TURN_SIGNAL_RIGHT = 2;
  TURN_SIGNAL_EMERGENCY = 3;
}

enum TurnSignalReason {
  TURN_SIGNAL_OFF = 0;
  LANE_CHANGE_TURN_SIGNAL = 1;
  TURNING_TURN_SIGNAL = 2;
  STARTING_TURN_SIGNAL = 3;
  PULL_OVER_TURN_SIGNAL = 4;
  TELEOP_TURN_SIGNAL = 5;
  MAP_DICTATED_TURN_SIGNAL = 6;
  FORK_TURN_SIGNAL = 7;
  MERGE_TURN_SIGNAL = 8;
  PREPARE_LANE_CHANGE_TURN_SIGNAL = 9;
  CONTINUE_LANE_CHANGE_SIGNAL = 10;
}